images/download_kafka.sh

```diff
@@ -20,20 +20,34 @@ SCALA_VERSION=$(cat supported_scala.version)
 #####
 for KAFKA_VERSION in $KAFKA_VERSIONS
 do
-    echo "Downloading: "$KAFKA_VERSION" with $SCALA_VERSION."
+
+    mkdir -p $KAFKA_DOWNLOADED_TARS_DIR
     KAFKA_URL="https://dlcdn.apache.org/kafka/$KAFKA_VERSION/kafka_$SCALA_VERSION-$KAFKA_VERSION.tgz"
     echo $KAFKA_URL
-    mkdir -p $KAFKA_DOWNLOADED_TARS_DIR
+    echo "Downloading: "$KAFKA_VERSION" with $SCALA_VERSION."
 
     if wget $KAFKA_URL -P "$KAFKA_DOWNLOADED_TARS_DIR"
     then
-      echo "We successfully download Kafka from $KAFKA_URL"
+        echo "We successfully download Kafka from $KAFKA_URL"
     else
-      echo "There was a problem downloading Kafka from $KAFKA_URL, we are gonna download it from archives."
-      KAFKA_URL="https://archive.apache.org/dist/kafka/$KAFKA_VERSION/kafka_$SCALA_VERSION-$KAFKA_VERSION.tgz"
-      wget $KAFKA_URL -P "$KAFKA_DOWNLOADED_TARS_DIR"
+        echo "There was a problem downloading Kafka from $KAFKA_URL, we are gonna download it from archives."
+        KAFKA_URL="https://archive.apache.org/dist/kafka/$KAFKA_VERSION/kafka_$SCALA_VERSION-$KAFKA_VERSION.tgz"
+
+        if wget $KAFKA_URL -P "$KAFKA_DOWNLOADED_TARS_DIR"
+        then
+            echo "We successfully download Kafka from $KAFKA_URL"
+        else
+            echo "There was a problem downloading Kafka from $KAFKA_URL, we are gonna download it from release candidate."
+            KAFKA_URL="https://home.apache.org/~satishd/kafka-$KAFKA_VERSION-rc0/kafka_2.13-$KAFKA_VERSION.tgz"
```

Comment:
This address will differ for every release. So I do not think this should be hardcoded in the code.